buildkite/resource_pipeline.go

```diff
@@ -61,479 +71,570 @@ type TeamPipelineNode struct {
 			},
-			"team": {
-				Type:       schema.TypeSet,
-				Optional:   true,
-				ConfigMode: schema.SchemaConfigModeAttr,
-				Elem: &schema.Resource{
-					Schema: map[string]*schema.Schema{
-						"slug": {
+			"webhook_url": schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"badge_url": schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+		},
+		Blocks: map[string]schema.Block{
+			"team": schema.SetNestedBlock{
+				NestedObject: schema.NestedBlockObject{
+					Attributes: map[string]schema.Attribute{
+						"slug": schema.StringAttribute{
 							Required: true,
-							Type:     schema.TypeString,
 						},
-						"access_level": {
+						"access_level": schema.StringAttribute{
 							Required: true,
-							Type:     schema.TypeString,
-							ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
-								switch v := val.(string); v {
-								case "READ_ONLY":
-								case "BUILD_AND_READ":
-								case "MANAGE_BUILD_AND_READ":
-									return
-								default:
-									errs = append(errs, fmt.Errorf("%q must be one of READ_ONLY, BUILD_AND_READ or MANAGE_BUILD_AND_READ, got: %s", key, v))
-									return
-								}
-								return
+							Validators: []validator.String{
+								stringvalidator.OneOf(string(PipelineAccessLevelsReadOnly), string(PipelineAccessLevelsBuildAndRead), string(PipelineAccessLevelsManageBuildAndRead)),
 							},
 						},
+						"team_id": schema.StringAttribute{
+							Computed: true,
+						},
+						"pipeline_team_id": schema.StringAttribute{
+							Computed: true,
+						},
 					},
 				},
 			},
-			"tags": {
-				Type:     schema.TypeSet,
-				Optional: true,
-				Elem:     &schema.Schema{Type: schema.TypeString},
-				Set:      schema.HashString,
-			},
-			"provider_settings": {
-				Type:     schema.TypeList,
-				Optional: true,
-				Computed: true,
-				MaxItems: 1,
-				Elem: &schema.Resource{
-					Schema: map[string]*schema.Schema{
-						"trigger_mode": {
+			"provider_settings": schema.ListNestedBlock{
+				NestedObject: schema.NestedBlockObject{
+					Attributes: map[string]schema.Attribute{
+						"trigger_mode": schema.StringAttribute{
 							Computed: true,
 							Optional: true,
-							Type:     schema.TypeString,
-							ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
-								switch v := val.(string); v {
-								case "code":
-								case "deployment":
-								case "fork":
-								case "none":
-									return
-								default:
-									errs = append(errs, fmt.Errorf("%q must be one of code, deployment, fork or none, got: %s", key, v))
-									return
-								}
-								return
+							Validators: []validator.String{
+								stringvalidator.OneOf("code", "deployment", "fork", "none"),
```

Comment:
There **must** be a way to stop people filling the settings if not `code`

Reply:
Yeah maybe if we write our own validator. But with this PR I want to mimic as close as possible (exactly) the SDKv2 behaviour. We can revisit improving this block when resolving #194